pisek/jobs/task_pipeline.py

```diff
@@ -47,53 +45,54 @@ def __init__(self, env: Env):
         super().__init__()
         named_pipeline = [
             tools := (ToolsManager(), TOOLS_MAN_CODE),
-            generator := (GeneratorManager(), GENERATOR_MAN_CODE),
+            generator := (PrepareGenerator(), GENERATOR_MAN_CODE),
+            checker := (CheckerManager(), CHECKER_MAN_CODE),
             inputs := (DataManager(), INPUTS_MAN_CODE),
         ]
         generator[0].add_prerequisite(*tools)
+        checker[0].add_prerequisite(*tools)
         inputs[0].add_prerequisite(*generator)
+        inputs[0].add_prerequisite(*checker)
 
-        if env.target != "solution":
-            named_pipeline.append(checker := (CheckerManager(), CHECKER_MAN_CODE))
-            checker[0].add_prerequisite(*inputs)
+        if env.target == TestingTarget.generator:
+            named_pipeline.append(gen_inputs := (RunGenerator(), ""))
+            gen_inputs[0].add_prerequisite(*inputs)
 
-        solutions = []
-        if env.solutions:
+        else:
             named_pipeline.append(judge := (JudgeManager(), JUDGE_MAN_CODE))
             judge[0].add_prerequisite(*inputs)
 
-            if env.config.judge_needs_out or (
-                env.config.primary_solution in env.solutions
-            ):
-                named_pipeline.append(
-                    primary_solution := (
-                        SolutionManager(env.config.primary_solution),
-                        f"{SOLUTION_MAN_CODE}{env.config.primary_solution}",
-                    )
-                )
-                solutions.append(primary_solution)
+            first_solution_name = (
+                env.config.primary_solution
+                if env.config.judge_needs_out
+                or env.config.primary_solution in env.solutions
+                else env.solutions[0]
+            )
```

Comment:
Why does the first solution need to be always handled differently?

Reply:
It generates the inputs, so other solutions depend on it.

Reply:
Wouldn't it be easier for `TaskPipeline` to create the generation job instead of having `TaskPipeline` pick a random different job to task with creating them?

Reply:
You mean adding ``RunGenerator()`` there?

Reply:
Yes

Reply:
This way we can run solution straight away which means you do not have to wait for generator to generate all inputs.

Reply:
Why does that require that the generator jobs are created by a solution?

Reply:
We want to run jobs this way:
- Generate input
- Check input
- Run solution
- Judge output
- Check output

We can change this manager to have a different name and display message, but it's still going to be manager of the first solution at the end of the day.

Reply:
If were running jobs in creation order, sure.
It might be cleaner to have each solution job try to find the correct generator jobs and create them if they don't exist, but that could be a lot of work.

Why is creating jobs so resource intensive anyway?